ha/util/consul_kv_store.py

```diff
@@ -176,7 +235,27 @@ def delete(self, key: str = "", recurse: bool = False):
             str: Return dictionary of all key val pair.
         """
         data = self.get(key)
-        self._consul.kv.delete(self._prepare_key(key), recurse=recurse)
+        k = self._prepare_key(key)
+
+        # if enabled batch put then delete from payload if any new value exist
```

Comment:
1. if recurse is True then there are many keys some are in payload and some are already in consul so needs to delete both.
2. in case if recurse is False the new value is in cache but there can be old value for same key is exist in consul needs to be deleted otherwise it will work like undo update and not delete.

Reply:
Addressed